### PR TITLE
Issue 12: Initial implementation of RS256 algorithm

### PR DESCRIPTION
**Pull request by jotsif**
RS256 implementation

### CODE REVIEW

File: src/test/scala/io/igl/jwt/JwtSpec.scala [hunk 1]
```diff
@@ -126,6 +126,21 @@ class JwtSpec extends UnitSpec {
...
+    val alg = Alg(Algorithm.RS256)
+    val pkbase64 = Base64.decodeBase64(pk)
```
Comment (iain-logan) [inline src/test/scala/io/igl/jwt/JwtSpec.scala]: I'm not so keen on the names `pk` and `pkbase64`, as we've just applied `decodeBase64` to _decode_ the base 64 encoded private key.

Basically, the variable names are the wrong way round.

Don't be afraid to use `privateKey` over `pk` as a name, it makes it more readable.
Short names don't buy us anything here.

File: src/test/scala/io/igl/jwt/JwtSpec.scala [hunk 2]
```diff
@@ -126,6 +126,21 @@ class JwtSpec extends UnitSpec {
...
+    val jwt=  new DecodedJwt(Seq(alg, Typ("JWT")), Seq(Iss("foo")))
+    val encoded = jwt.encodedAndSigned(pkbase64)
+    DecodedJwt.validateEncodedJwtWithEncodedSecret(
```
Comment (iain-logan) [inline src/test/scala/io/igl/jwt/JwtSpec.scala]: Super picky here, but please add a blank line between declaring the values and validating the JWT to match the other tests.

**Review by iain-logan — CHANGES_REQUESTED** [review]
Really appreciate the pull request, just a few tiny stylistic changes I'd like you to make. Thanks!

**Comment by jotsif** [comment]
Variables renamed and newline added.

**Comment by iain-logan** [comment]
Thanks, pull request accepted

**Comment by jotsif** [comment]
Thanks @iain-logan !

**Pull request merged**